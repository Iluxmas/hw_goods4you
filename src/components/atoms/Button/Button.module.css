.root {
  padding: 22px 47px;
  border-radius: 4px;
  border: 1px solid transparent;
  background-color: transparent;
  white-space: nowrap;
  font-size: 1rem;
  transition: opacity 0.2s ease;

  &:hover {
    cursor: pointer;
    opacity: 0.8;
  }

  &.primary {
    background-color: var(--brand-red);
    color: var(--white);
  }

  &.secondary {
    color: var(--white);
    background-color: var(--dark-gray);
  }

  &.outline {
    color: var(--dark-gray);
    border: 1px solid var(--dark-gray);
  }

  &.link {
    padding: 0;
    border: none;
    color: var(--dark-gray);
    transition: all 0.2s ease;

    &:hover {
      text-decoration: underline;
    }
  }
}
