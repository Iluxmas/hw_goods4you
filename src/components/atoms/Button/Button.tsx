import clsx from 'clsx';
import Loader from '@atoms/loader/Loader';

import s from './Button.module.css';

type Props = {
  text: string;
  disabled?: boolean;
  className?: string;
  isLoading?: boolean;
  onClick?: () => void;
  intent?: 'primary' | 'secondary' | 'outline' | 'link';
};

function Button({
  text,
  onClick,
  disabled,
  className,
  isLoading,
  intent = 'primary',
}: Props) {
  return (
    <button
      type="button"
      disabled={disabled || isLoading}
      className={clsx(
        s.root,
        s[intent],
        className,
        disabled && s.disabled,
        isLoading && s.loading
      )}
      onClick={onClick}>
      {isLoading ? <Loader size={18} color="#b2b5bb" /> : text}
    </button>
  );
}

export default Button;
