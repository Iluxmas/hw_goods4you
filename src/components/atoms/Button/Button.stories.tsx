import { Meta, StoryObj } from '@storybook/react';
import Button from './Button';
import '../../../index.css';

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    intent: {
      type: 'string',
      description: 'style',
      defaultValue: 'primary',
      options: ['primary', 'secondary', 'link', 'outline'],
      control: {
        type: 'radio',
      },
    },
  },
} as Meta<typeof Button>;

type Story = StoryObj<typeof Button>;

export const Default = {
  args: { text: 'Click Me', intent: 'primary' },
} satisfies Story;
