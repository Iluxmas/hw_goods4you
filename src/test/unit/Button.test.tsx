import { it, expect } from '@jest/globals';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Button from '../../components/atoms/Button/Button';

describe('\nðŸ”µ Button should work', function () {
  it('Should change the class with different isloading/disabled props', () => {
    const component = renderer.create(
      <Button text="Test me" disabled={false} isLoading={false} />
    );

    let tree = component.toJSON() as ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
      component.update(
        <Button text="Test me" disabled={true} isLoading={false} />
      );
    });

    tree = component.toJSON() as ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
  });

  it('Should have passed text', () => {
    const { rerender } = render(<Button text="Test me" />);
    expect(screen.getByTestId('button-test-id').textContent).toBe('Test me');

    rerender(<Button text="Other Text" />);
    expect(screen.getByTestId('button-test-id').textContent).toBe('Other Text');
  });
});
