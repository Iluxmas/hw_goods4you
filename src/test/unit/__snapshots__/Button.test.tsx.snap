// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`
ðŸ”µ Button should work Should change the class with different isloading/disabled props 1`] = `
<button
  className="root primary"
  data-testid="button-test-id"
  disabled={false}
  type="button"
>
  Test me
</button>
`;

exports[`
ðŸ”µ Button should work Should change the class with different isloading/disabled props 2`] = `
<button
  className="root primary disabled"
  data-testid="button-test-id"
  disabled={true}
  type="button"
>
  Test me
</button>
`;
